packages/service-provider-core/src/connection-string.ts

```diff
@@ -0,0 +1,130 @@
+import { CommonErrors, MongoshInvalidInputError, MongoshInternalError } from '@mongosh/errors';
+// The java-shell doesn't have URL, so we fall back to a pure-JS implementation.
+// And, because it's so much fun, it also doesn't have TextEncoder/TextDecoder,
+// so we need to (crudely) polyfill that as well in order to use that
+// pure-JS implementation.
+if (typeof require('util').TextDecoder !== 'function' || typeof require('util').TextEncoder !== 'function') {
+  class TextEncoder {
+    encode(string: string): Uint8Array { return Buffer.from(string, 'utf8'); }
+  }
+  class TextDecoder {
+    decode(bytes: Uint8Array): string {
+      const str = Buffer.from(bytes).toString('utf8');
+      return str.slice(+str.startsWith('\ufeff'));
+    }
+  }
+  Object.assign(require('util'), { TextDecoder, TextEncoder });
+}
+import { URL as WhatWGURL } from 'whatwg-url';
```

Comment:
Is the url from browserify not compliant?

Reply:
The problem is that there isn’t any :grimacing: